### PR TITLE
sdk/log: Fix counting number of dropped attributes of log `Record`

### PR DESCRIPTION
**Pull request by amanakin**
Applying attribute limits in `Record` uses value receiver.
But it should add count of dropped attrs: 
https://github.com/open-telemetry/opentelemetry-go/blob/main/sdk/log/record.go#L416
In this PR I add using of pointer receiver.

Also it's slightly faster with pointer receiver:
```
goos: darwin
goarch: arm64
pkg: go.opentelemetry.io/otel/sdk/log
                    │   old.txt   │              new.txt               │
                    │   sec/op    │   sec/op     vs base               │
SetAddAttributes-10   175.7n ± 3%   159.8n ± 4%  -9.08% (p=0.000 n=10)

                    │  old.txt   │            new.txt             │
                    │    B/op    │    B/op     vs base            │
SetAddAttributes-10   48.00 ± 0%   48.00 ± 0%  ~ (p=1.000 n=10) ¹
¹ all samples are equal

                    │  old.txt   │            new.txt             │
                    │ allocs/op  │ allocs/op   vs base            │
SetAddAttributes-10   1.000 ± 0%   1.000 ± 0%  ~ (p=1.000 n=10) ¹
¹ all samples are equal
```

### CODE REVIEW

**Comment by dmathieu** [comment]
Since this is fixing an issue counting the number of dropped attributes, could you add a changelog entry?

**Comment by amanakin** [comment]
@dmathieu Sure, added changelog entry.

**Comment by MrAlias** [comment]
> **Before**: `BenchmarkSetAttributes-10 11057461 102.5 ns/op 48 B/op 1 allocs/op`
> **After**: `BenchmarkSetAttributes-10 12738790 86.69 ns/op 48 B/op 1 allocs/op`

Please use [benchstat](https://pkg.go.dev/golang.org/x/perf/cmd/benchstat) to report the statistical significance of these benchmark changes.

File: sdk/log/record_test.go [hunk 1]
```diff
@@ -431,11 +434,13 @@ func TestApplyAttrLimitsDeduplication(t *testing.T) {
 			t.Run("AddAttributes", func(t *testing.T) {
 				r.AddAttributes(kv)
 				assertKV(t, r, log.KeyValue{Key: key, Value: tc.want})
+				require.Equal(t, tc.droppedAttrs, r.DroppedAttributes())
```
Comment (pellared) [inline sdk/log/record_test.go:437]:
```suggestion
				assert.Equal(t, tc.droppedAttrs, r.DroppedAttributes())
```

File: sdk/log/record_test.go [hunk 2]
```diff
@@ -431,11 +434,13 @@ func TestApplyAttrLimitsDeduplication(t *testing.T) {
...
 			})
 
 			t.Run("SetAttributes", func(t *testing.T) {
 				r.SetAttributes(kv)
 				assertKV(t, r, log.KeyValue{Key: key, Value: tc.want})
+				require.Equal(t, tc.droppedAttrs, r.DroppedAttributes())
```
Comment (pellared) [inline sdk/log/record_test.go:443]:
```suggestion
				assert.Equal(t, tc.droppedAttrs, r.DroppedAttributes())
```

File: sdk/log/record_test.go [hunk 3]
```diff
@@ -349,6 +349,8 @@ func TestApplyAttrLimitsDeduplication(t *testing.T) {
 		name        string
 		limit       int
 		input, want log.Value
+
+		droppedAttrs int
```
Comment (pellared) [inline sdk/log/record_test.go:353]:
```suggestion
		wantDroppedAttrs int
```

File: sdk/log/record_test.go [hunk 4]
```diff
@@ -635,3 +640,12 @@ func TestTruncate(t *testing.T) {
...
 	}
 }
+
+func BenchmarkSetAddAttributes(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		kv := log.String("key", "value")
+		var r Record
+		r.SetAttributes(kv)
+		r.AddAttributes(kv)
+	}
```
Comment (MrAlias) [inline sdk/log/record_test.go:650]: The `kv` and `r` declarations should not be included in the benchmark, they are unrelated to the tested methods. Also, you are showing memory benchmarks, but this does not output any memory benchmarks.

```suggestion
	kv := log.String("key", "value")
	records := make([]Record, b.N)

	b.ReportAllocs()
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		records[i].SetAttributes(kv)
		records[i].AddAttributes(kv)
	}
```

**Review by pellared — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

**Comment by pellared** [comment]
@amanakin, thank you for your contribution 🎉

**Pull request merged**